rosbag2_cpp/include/rosbag2_cpp/compression_options.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_CPP__COMPRESSION_OPTIONS_HPP_
+#define ROSBAG2_CPP__COMPRESSION_OPTIONS_HPP_
+
+#include <string>
+
+namespace rosbag2_cpp
+{
+
+/**
+ * Modes are used to specify whether to compress by individual serialized bag messages or by file.
+ * rosbag2_cpp defaults to NONE.
+ */
+enum class CompressionMode
+{
+  NONE,
+  FILE,
+  MESSAGE,
+};
```

Comment:
It might be better to do:
```c++
enum class CompressionMode : uint32_t
{
  NONE = 0,
  FILE,
  MESSAGE,
  LAST_MODE = MESSAGE
};
```
since one of the unit tests is doing a `static_cast<CompressionMode>` on an integer-type.
You can then make the test more robust by doing: 
```c++
auto bad_mode = static_cast<CompressionMode>(static_cast<uint32_t>(CompressionMode::LAST_MODE) + 1);
```